### PR TITLE
fixed 4anime & added an unpacker for packed js that uses pure python

### PR DESCRIPTION
**Pull request by ArjixGamer**
needs testing

### CODE REVIEW

**Comment by ArjixGamer** [comment]
i did test it, but i can never be 100% sure so go wild testing

**Comment by ArjixGamer** [comment]
too bad that more and more providers depend on nodejs...
maybe we should write our own unpacker for ``p.a.c.k.e.d.`` js

**Comment by ArjixGamer** [comment]
found this beauty: https://github.com/beautify-web/js-beautify/blob/main/python/jsbeautifier/unpackers/packer.py
will do some tests on it

**Comment by ArjixGamer** [comment]
aight, now 4anime doesnt use nodejs at all

**Comment by ArjixGamer** [comment]
the unpacker can be used like this:
```python
from anime_downloader.sites import helpers

if helpers.detect_packed_js(packed_js_string):
    result = helpers.unpack_packed_js(packed_js_string)
```

**Comment by ArjixGamer** [comment]
changed it to import the unpackers instead of just copy pasting their code in anime dl

**Comment by ArjixGamer** [comment]
this pr once again works

File: setup.py [hunk 1]
```diff
@@ -31,6 +31,7 @@
         'requests-cache>=0.4.13',
         'tabulate>=0.8.3',
         'pycryptodome>=3.8.2',
+        'jsbeautifier==1.11.0'
```
Comment (AbdullahM0hamed) [inline setup.py:34]: Doing this apparently has the potential to break this for some AUR and brew users. Perhaps move this to `extras_require` (add to existing, and one of its own)?

Comment (ArjixGamer) [reply setup.py:34]: how tf does it have such a potential tho?
its not our problem if aur is fucked up, is it?

Comment (ArjixGamer) [reply setup.py:34]: in the first place we dont the aur package is an unofficial one...

Comment (AbdullahM0hamed) [reply setup.py:34]: It's unofficial? Strange that vn-ki would comment about it (see: https://github.com/anime-dl/anime-downloader/pull/443#issuecomment-662447368)

Comment (ArjixGamer) [reply setup.py:34]: ok fine, you win, ill make it an optional dependency

**Comment by ArjixGamer** [comment]
Ok, so I finished testing.
The unpacker returns its own input if jsbeautifier is not installed.
That way, it has the same result as it not being there at all.

**Pull request merged**